src/main/assisted-installer-controller/assisted_installer_main.go

```diff
@@ -200,15 +200,22 @@ func main() {
 	case assistedinstallercontroller.InvokerAgent:
 		waitForInstallationAgentBasedInstaller(kc, logger, removeUninitializedTaint)
 	default:
-		waitForInstallation(client, logger, assistedController.Status)
+		waitForInstallation(client, logger, assistedController)
+	}
+}
+
+func checkForTimedOutFinalizingStage(cluster *models.Cluster, controller assistedinstallercontroller.Controller, log logrus.FieldLogger) {
+	if cluster != nil && cluster.Progress != nil && cluster.Progress.FinalizingStageTimedOut {
```

Comment:
Something i am missing here, cluster is not in error state? Why we verify progress and not final cluster stage?